cms/src/signed_data.rs

```diff
@@ -181,6 +181,21 @@ impl ValueOrd for SignerIdentifier {
     }
 }
 
+impl From<Certificate> for SignerIdentifier {
+    fn from(cert: Certificate) -> Self {
+        let tbs = cert.tbs_certificate();
+        if let Ok(Some((_critical, ski))) = tbs.get_extension::<SubjectKeyIdentifier>() {
+            Self::SubjectKeyIdentifier(ski)
+        } else {
+            let isn = IssuerAndSerialNumber {
+                issuer: tbs.issuer().clone(),
+                serial_number: tbs.serial_number().clone(),
+            };
+            Self::IssuerAndSerialNumber(isn)
+        }
```

Comment:
This feels like it might look better as a match, e.g.:

```suggestion
        match tbs.get_extension::<SubjectKeyIdentifier>() {
            Ok(Some((_critical, ski))) => Self::SubjectKeyIdentifier(ski),
            _ => Self::IssuerAndSerialNumber(IssuerAndSerialNumber {
                issuer: tbs.issuer().clone(),
                serial_number: tbs.serial_number().clone(),
            })
        }
```